homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+                "version": device["version"],
+                "status": device["status"],
+                "createTime": device["userDeviceCreateTime"],
+                "category": device["deviceCategory"],
+                "subCategory": device["deviceSubCategory"],
+                "customType": device["customType"],
+                "localIp": connections[device_serial]["localIp"],
+                "netIp": connections[device_serial]["netIp"],
+                "localRtspPort": local_rtsp_port,
+                "netType": connections[device_serial]["netType"],
+                "wanIp": connections[device_serial]["wanIp"],
+            }
+
+        # Add the cameras as devices in HASS
+        for camera_serial in cameras:
+            camera = cameras[camera_serial]
+            _LOGGER.debug("CAMERA: %s", camera)
+
+            camera_username = CONF_DEFAULT_CAMERA_USERNAME
+            camera_password = ""
+            camera_rtsp_stream = ""
+
+            if camera_serial in conf_cameras:
+                camera_username = conf_cameras[camera_serial]["username"]
+                camera_password = conf_cameras[camera_serial]["password"]
+                camera_rtsp_stream = "rtsp://{}:{}@{}:{}".format(
+                    camera_username,
+                    camera_password,
+                    camera["localIp"],
+                    camera["localRtspPort"],
+                )
+                _LOGGER.debug(
+                    "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+                )
+
+            else:
+                _LOGGER.error(
+                    "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
+                    camera_serial,
+                    conf_cameras,
+                )
+
+            async_add_entities(
+                [
+                    EzvizCamera(
+                        hass,
+                        camera_username,
+                        camera_password,
+                        camera_rtsp_stream,
+                        camera["serial"],
+                        camera["name"],
+                        camera["deviceType"],
+                        camera["version"],
+                        camera["status"],
+                        camera["createTime"],
+                        camera["category"],
+                        camera["subCategory"],
+                        camera["customType"],
+                        camera["localIp"],
+                        camera["netIp"],
+                        camera["localRtspPort"],
+                        camera["wanIp"],
+                        camera["netType"],
+                    )
+                ]
+            )
+    except PyEzvizError as exp:
+        _LOGGER.error(exp)
+        return
+
+
+class EzvizCamera(Camera):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(
+        self,
+        hass,
```

Comment:
Do not pass `hass`, it is populated as `self.hass` when the device is added to HA during `add_devices`

Reply:
Didn't know that, thanks for the advice!